src/d/actor/d_a_player_rope.inc

```diff
@@ -164,96 +427,809 @@ BOOL daPy_lk_c::procRopeSubject() {
+    }
+LAB_801443c4:
+    m3600 = m370C.y;
+    m370C.z = pfVar3->current.pos.z;
+    gravity = 0.0f;
+    mVelocity = 0.0f;
+    m35A0 = 0.0f;
+    fopAcM_SetParam(pfVar3, 3);
+    m34D4 = 0x1800;
+    shape_angle.y = cM_atan2s((m370C.x - current.pos.x), (m370C.z - current.pos.z));
+    current.angle.y = shape_angle.y;
+    mRopePos = pfVar3->current.pos;
+    dComIfGp_setPlayerStatus0(0, daPyStts0_UNK800000_e);
+    return true;
 }
 
 /* 80144464-80144620       .text procRopeReady__9daPy_lk_cFv */
 BOOL daPy_lk_c::procRopeReady() {
-    /* Nonmatching */
+    /* Nonmatching - floats */
+    if (!checkItemModeActorPointer()) {
+        return true;
+    }
+    fopAc_ac_c* pfVar2 = mActorKeepEquip.getActor();
+    mRopePos = pfVar2->current.pos;
+    m370C.x = mRopePos.x;
+    m370C.z = mRopePos.z;
+    cXyz local_48 = (m370C - current.pos);
+    f32 dVar5 = local_48.abs();
+    cLib_chaseF(&m35A0, 50.0f, 5.0f);
+    shape_angle.x = (0.025f * (m34D4 * m35A0));
+    if (dVar5 > m35A0) {
+        current.pos += local_48 * (m35A0 / dVar5);
+    }
+    if (dVar5 <= (2.0f * m35A0)) {
+        procRopeSwing_init(mActorKeepEquip.getActor(), shape_angle.x);
+    }
+    return true;
 }
 
 /* 80144620-80144C58       .text procRopeSwing_init__9daPy_lk_cFP10fopAc_ac_cs */
-BOOL daPy_lk_c::procRopeSwing_init(fopAc_ac_c*, s16) {
-    /* Nonmatching */
+BOOL daPy_lk_c::procRopeSwing_init(fopAc_ac_c* param_0, s16 param_1) {
+    /* Nonmatching - floats, regalloc */
+    float dVar8;
+    float dVar9;
+    cXyz local_84;
+    cXyz local_78;
+
+    himo2_class* pfVar7 = (himo2_class*)param_0;
+    commonProcInit(daPyProc_ROPE_SWING_e);
+    if (param_0 != NULL) {
+        mActorKeepRope.setData(param_0);
+        if (fopAcM_GetName(param_0) == PROC_HIMO3) {
+            fopAcM_SetParam(param_0, 1);
+            deleteEquipItem(TRUE);
+            mEquipItem = dItem_GRAPPLING_HOOK_e;
+            local_84 = (param_0->current.pos - current.pos);
+            dVar8 = (local_84.abs() - 95.0f);
+            dVar9 = pfVar7->field_0x15F8.y;
+            if (dVar8 > dVar9) {
+                dVar8 = dVar9;
+            }
+            local_84.normalize();
+            current.pos = (param_0->current.pos - (local_84 * dVar8));
+        }
+    } else {
+        param_0 = mActorKeepRope.getActor();
+    }
+    local_84 = (param_0->current.pos - current.pos);
+    m35A0 = local_84.abs();
+    f32 dVar11 = std::sqrtf(2.0f / m35A0);
+    m35A4 = dVar11;
+    dVar11 = (M_PI / 2 / m35A4);
+    if (pfVar7 != NULL) {
+        m34D4 = daPy_HIO_rope_c0::m.field_0x0;
+        if (checkSpecialRope()) {
+            m34D4 *= 0.125f;
+        }
+        f32 fVar1 = cM_ssin(shape_angle.y);
+        f32 fVar2 = cM_scos(shape_angle.y);
+        local_78.x = fVar2 * local_84.x - fVar1 * local_84.z;
+        local_78.y = local_84.y;
+        local_78.z = fVar1 * local_84.x + fVar2 * local_84.z;
+        s16 sVar4 = cM_atan2s(-local_78.x, local_78.z);
+        if (!(sVar4 <= m34D4)) {
```

Comment:
```suggestion
        if (sVar4 > m34D4) {
```
You can clean up negated conditions by just inverting them (only if it's an integer comparison, floats are more complicated)